pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,477 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	viper "github.com/openshift/osde2e/pkg/common/concurrentviper"
+	"github.com/openshift/osde2e/pkg/common/config"
+	"github.com/openshift/osde2e/pkg/common/expect"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/common/label"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+const (
+	nodePoolLabel = "hypershift.openshift.io/nodePool"
+)
+
+var nodePoolGVR = schema.GroupVersionResource{
+	Group:    "hypershift.openshift.io",
+	Version:  "v1beta1",
+	Resource: "nodepools",
+}
+
+var _ = ginkgo.Describe("[Suite: e2e] NodePool STS Permissions", ginkgo.Ordered, label.HyperShift, label.E2E, func() {
+	var (
+		h                *helper.H
+		client           *resources.Resources
+		clusterNamespace string
+		testNodePoolName string
+		createdNodePool  bool
+	)
+
+	ginkgo.BeforeAll(func() {
+		cloudProvider := viper.GetString(config.CloudProvider.CloudProviderID)
+		if cloudProvider != "aws" {
+			ginkgo.Skip(fmt.Sprintf("Tests only supported on AWS, got %s", cloudProvider))
+		}
+
+		h = helper.New()
+		client = h.AsUser("")
+
+		var err error
+		clusterNamespace, err = getClusterNamespace(h)
+		if err != nil {
+			ginkgo.Skip(fmt.Sprintf("Cannot determine cluster namespace: %v", err))
+		}
+
+		testNodePoolName = fmt.Sprintf("sts-test-%d", time.Now().Unix()%100000)
+	})
+
+	ginkgo.AfterAll(func() {
+		if createdNodePool && testNodePoolName != "" && clusterNamespace != "" {
+			cleanupNodePool(context.Background(), h, clusterNamespace, testNodePoolName)
+		}
+	})
+
+	ginkgo.It("validates existing nodes have proper AWS integration", func(ctx context.Context) {
+		nodePoolNodes, err := findNodePoolNodes(ctx, client)
+		if isSTSError(err) {
+			failWithSTSError(err, "node discovery", "ec2:DescribeInstances")
+		}
+		expect.NoError(err, "Failed to find NodePool nodes")
+		Expect(len(nodePoolNodes)).To(BeNumerically(">", 0), "No NodePool nodes found")
+
+		for _, node := range nodePoolNodes {
+			validateAWSIntegration(node)
+		}
+	})
+
+	ginkgo.It("creates new NodePool successfully", func(ctx context.Context) {
+		if clusterNamespace == "" {
```

Comment:
Once here, clusterNamespace should never by an empty string, should it? And, can the canAccessNodePoolAPI test be done before any tests are run?

Note, this isn't consistent with the others.